c7n/policy.py

```diff
@@ -73,8 +75,13 @@ def __init__(self, data, options):
         # We store all the policies passed in so we can refilter later
         self._all_policies = []
         for p in self.data.get('policies', []):
-            self._all_policies.append(
-                Policy(p, options, session_factory=self.test_session_factory()))
+            if 'all' in options.regions:
+                options.regions = utils.get_available_regions(p['resource'])
+            for region in options.regions:
+                options_copy = copy.copy(options)
+                options_copy.region = region
+                self._all_policies.append(
```

Comment:
ideally we don't create and store copies for invalid policy resources regions, part of the notion of filtering out policies automatically based on invalid regions for that policy resource, ie this is only filtering on the all. if we're explicitly told to run a policy against region where it doesn't exist we should also log a warning and skip it, since it can't run.